### PR TITLE
mail module: add Message-ID header

### PR DESCRIPTION
**Pull request by sbocahu**
##### SUMMARY

Add Message-ID header to emails sent with the mail module (mandatory to be accepted by some mail servers; such as Gmail)

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

**Comment by sbocahu** [comment]
@felixfontein done ✔️

File: plugins/modules/mail.py [hunk 1]
```diff
@@ -348,6 +348,7 @@ def main():
     msg['From'] = formataddr((sender_phrase, sender_addr))
     msg['Date'] = formatdate(localtime=True)
     msg['Subject'] = Header(subject, charset)
+    msg['Message-ID'] = make_msgid()
```
Comment (parsa97) [inline plugins/modules/mail.py:351]: AFIK the Message-ID is a string like `<170301148900.46620.6704225022053582822@Domain>` and the `make_msgid` will expose the localhost machine name by default. I think we have to use sth like `msgid_domainname` to cusotm this domain name.

```suggestion
    if msgid_domainname is not None:
      msg['Message-ID'] = make_msgid(domain=msgid_domainname)
```

**Review by parsa97 — CHANGES_REQUESTED** [review]

File: plugins/modules/mail.py (hunk 1 above)
Comment (felixfontein) [reply plugins/modules/mail.py:351]: While such a configuration option would certainly be nice, I don't think it's absolutely necessary. The `mail` callback plugin in this collection also simply uses `make_msgid()` to create a message ID. Also most MUAs also simply use localhost's machine name by default.

Comment (parsa97) [reply plugins/modules/mail.py:351]: I have security concerns about the default functionality because this will expose our ansible machine hostname.

Comment (felixfontein) [reply plugins/modules/mail.py:351]: A better solution (for now) would be
```suggestion
    msg['Message-ID'] = make_msgid(domain='ansible')
```
The next step would be making that value configurable, but that can happen in a follow-up PR.

File: changelogs/fragments/7740-add-message-id-header-to-mail-module.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - mail module - add Message-ID header; which is required by some mail servers (https://github.com/ansible-collections/community.general/pull/7740).
```
Comment (felixfontein) [inline changelogs/fragments/7740-add-message-id-header-to-mail-module.yml:2]:
```suggestion
  - mail - add ``Message-ID`` header; which is required by some mail servers (https://github.com/ansible-collections/community.general/pull/7740).
```

**Review by felixfontein — APPROVED** [review]
If nobody objects, I'll merge in a few days.

File: plugins/modules/mail.py [hunk 3]
```diff
@@ -348,6 +348,7 @@ def main():
     msg['From'] = formataddr((sender_phrase, sender_addr))
     msg['Date'] = formatdate(localtime=True)
     msg['Subject'] = Header(subject, charset)
+    msg['Message-ID'] = make_msgid(domain='ansible')
```
Comment (felixfontein) [inline plugins/modules/mail.py:351]: It looks like the `domain` option was only added in Python 3.2, and never backported to Python 2. Let's do the following:
```suggestion
    try:
        msg['Message-ID'] = make_msgid(domain='ansible')
    except TypeError:
        # `domain` is only available in Python 3
        msg['Message-ID'] = make_msgid()
```

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@sbocahu thanks for your contribution!
@parsa97 thanks for reviewing!

**Pull request merged**